File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int maxP = 0, buyStock = prices[0];

        for(int i=1; i<prices.size(); i++){
            if(prices[i] > buyStock){
                maxP = max(maxP , prices[i]-buyStock);
            }
            buyStock = min(prices[i],buyStock);
        }
        return maxP;
    }
};

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.js
/**
 * @param {number[]} prices
 * @return {number}
 */
var maxProfit = function(prices) {
    let maxProfit = 0
    let minPrice = Infinity

    for(let currPrice of prices){
        if(currPrice < minPrice){
            minPrice = currPrice
        }else{
            maxProfit = Math.max(maxProfit, currPrice - minPrice)
        }
       
    }
    return maxProfit 
};